sql/sql_show.cc

```diff
@@ -6819,6 +6819,42 @@ store_constraints(THD *thd, TABLE *table, const LEX_CSTRING *db_name,
   return schema_table_store_record(thd, table);
 }
 
+static int get_check_constraints_record(THD *thd, TABLE_LIST *tables,
+                                        TABLE *table, bool res,
+                                        const LEX_CSTRING *db_name,
+                                        const LEX_CSTRING *table_name)
+{
+  DBUG_ENTER("get_check_constraints_record");
+  if (res)
+  {
+    if (thd->is_error())
+      push_warning(thd, Sql_condition::WARN_LEVEL_WARN,
+                   thd->get_stmt_da()->sql_errno(),
+                   thd->get_stmt_da()->message());
+    thd->clear_error();
+    DBUG_RETURN(0);
+  }
+  else if (!tables->view)
+  {
+    if (tables->table->s->table_check_constraints)
+    {
+      for (uint i= 0; i < tables->table->s->table_check_constraints; i++)
+      {
+        StringBuffer<MAX_FIELD_WIDTH> str(system_charset_info);
```

Comment:
We are recreating the StringBuffer object on every for loop iteration. Although compiler might be smart enough to optimize this, let's do it ourselves. Move the object definition outside the for loop and just set the string's lenth to 0 before printing to it. When calling `check->print(&str);` make it:
```
str.length(0);
check->print(&str);
```

Reply:
Also, I see that you've changed this to system_charset_info. Looking at fb67cde2370f printing vcol was changed from using system_charset_info to my_charset_utf8mb4_general_ci. One needs to be able to answer the question as to why that was done there and why are we ok to keep system_charset_info here?